packages/amplify-appsync-simulator/src/server/subscription/websocket-server/server.ts

```diff
@@ -261,7 +266,7 @@ export class WebsocketSubscriptionServer {
 
   private attachAsyncIterator = async (connectionContext: ConnectionContext, sub: WebsocketSubscription): Promise<void> => {
     const { asyncIterator, id } = sub;
-    while (true) {
+    for (;;) {
```

Comment:
:-(

Reply:
The syntax is odd. It is idiomatic (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for#optional_for_expressions), but I agree that it is unusual enough to not be included. I refactored this function in my latest commit.

Reply:
see also: https://eslint.org/docs/latest/rules/no-constant-condition

Reply:
I know the `for (;;)` syntax. I just find it uglier than `while (true)`.

Reply:
btw. that particular piece of code looks like good candidate for `do {} while()` loop :-)

Reply:
Unfortunately, it's sixes. Because of the boolean check between the two operations, the `while` vs `do` are both a bit clunky.